packages/@react-stately/select/src/useSelectState.ts

```diff
@@ -44,34 +78,85 @@ export interface SelectState<T> extends SingleSelectListState<T>, OverlayTrigger
  * of items from props, handles the open state for the popup menu, and manages
  * multiple selection state.
  */
-export function useSelectState<T extends object>(props: SelectStateOptions<T>): SelectState<T>  {
+export function useSelectState<T extends object, M extends SelectionMode = 'single'>(props: SelectStateOptions<T, M>): SelectState<T, M>  {
+  let {selectionMode = 'single' as M} = props;
   let triggerState = useOverlayTriggerState(props);
   let [focusStrategy, setFocusStrategy] = useState<FocusStrategy | null>(null);
-  let listState = useSingleSelectListState({
+  let defaultValue = useMemo(() => {
+    return props.defaultValue ?? (selectionMode === 'single' ? props.defaultSelectedKey ?? null : []) as ValueType<M>;
+  }, [props.defaultValue, props.defaultSelectedKey, selectionMode]);
+  let value = useMemo(() => {
+    return props.value ?? (selectionMode === 'single' ? props.selectedKey : undefined) as ValueType<M>;
```

Comment:
```suggestion
    return props.value !== undefined ? value : (selectionMode === 'single' ? props.selectedKey : undefined) as ValueType<M>;
```

The original line causes an issue in which `value={null}` is regarded as `value={undefined}`, therefore the value cannot be forced cleared.